### PR TITLE
Product info

### PR DESCRIPTION
**Pull request by WildMountainAir**
Size Dropdown renders only available sizes. 
Quantity Dropdown renders 'select size' until a size is selected and then renders 1 to nQuantityInStock.
Price updates based on selected style.
Thumbnails now display selected style with a peach quartz circle.
Main carousel updates when a thumbnail is selected, defaulted to the default style of a product.

### CODE REVIEW

File: client/src/app.jsx [hunk 1]
```diff
@@ -33,14 +37,24 @@ const App = () => {
...
-  || Object.keys(styles).length === 0) {
+  || Object.keys(styles).length === 0
+  || Object.keys(currentStyle).length === 0
+  || reviews === undefined
+  || product === undefined
+  || styles === undefined
+  || currentStyle === undefined) {
     return (
       <div className="loading loading-img">
         <h1 className="loading-h1">The Proto Company</h1>
```
Comment (OhHeyDeer) [inline client/src/app.jsx:60]: Love this functionality for the loading page

File: client/src/components/sizeDropdown.jsx [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
       value="size"
+      onChange={(value) => handleSetSku(value)}
       placeholder="Select an option"
     />
   );
 };
 
+SizeDropdown.propTypes = {
+  currentStyle: PropTypes.shape({
+    'default?': PropTypes.number,
```
Comment (OhHeyDeer) [inline client/src/components/sizeDropdown.jsx:29]: Ooh this is cool! I was wondering how to set default values.

File: client/src/components/styleThumbnails.jsx [hunk 3]
```diff
@@ -1,47 +1,54 @@
...
-    imgIsSelected(!isSelected); // current style will go here
-  }, [isSelected]);
+const StyleThumbnails = ({ styles, setCurrentStyle, currentStyle }) => {
+  const stylesArr = Object.entries(styles.results);
 
-  let className = 'thumbnails-img ';
-  if (isSelected) {
-    className += 'thumbnails-selected';
-  }
+  const handleStyleSelect = (styleId) => {
```
Comment (OhHeyDeer) [inline client/src/components/styleThumbnails.jsx:10]: className might be a bit too vague.. Totally up to you tho!

**Review by OhHeyDeer — COMMENTED** [review]
Looks great!! Seems like there was a TON of progress!

File: client/src/components/styleThumbnails.jsx (hunk 3 above)
Comment (WildMountainAir) [reply client/src/components/styleThumbnails.jsx:10]: It is the classname for the className in the Image so I tried to just keep everything the same, but would love any recommendations for renaming! :D

**Pull request merged**